<h1 style = "color : #0EE071; text-align : center;"><em>Nata Project</em> - Data Exploration Notebook</h1>
<p style = "font-size : 16px; text-align: center;">This notebook is dedicated only to the exploration of the <code>Nata_Files/learn.csv</code> dataset</p>
<br>
<p style = "font-size : 12px; text-align: center;"><b>NOVA IMS</b></p>
<p style = "font-size : 10px; text-align: center;">Machine Learning I</p>
<p style = "font-size : 10px; text-align: center;">Diogo Gonçalves, João Marques, Juan Mendes, Gustavo Franco & Lucas Casimiro</p>
<br>


<h2  style = "color : #0EE071;"> Imports</h2>

In [23]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

<hr style = "border: 3px solid #0EE071;">
<h2 style = "color : #0EE071;">Dataset Reading </h2>
<p style = "font-size : 15px;">Reading of dataset from <code>learn.csv</code> file</p>
<p>The dataset is in a <code>.csv</code> file. We are also adopting the column <code>id</code> as index, thus <code>index_col = 0</code>. </p>

In [25]:
learn_data = pd.read_csv("Nata_Files/learn.csv", index_col = 0)

In [26]:
learn_data.head()

ambient_humidity  baking_duration  cooling_period  cream_fat_content  \
id                                                                         
1               54.0             24.0            26.0              100.4   
2               66.0             37.0            34.0               98.0   
3               41.0             30.0            19.0               99.3   
4               62.0             24.0            48.0               98.0   
5               55.0             21.0            34.0              100.1   

    egg_temperature  egg_yolk_count  final_temperature  lemon_zest_ph  \
id                                                                      
1              52.0            11.0              309.0            3.2   
2              46.0            10.0              317.0            3.3   
3              53.0            10.0              130.0            3.4   
4             115.0             9.0              354.0            3.3   
5              48.0             9.0              211.0            3.0   

    notes_baker   origin  oven_temperature     pastry_type  preheating_time  \
id                                                                            
1           NaN   Lisboa               NaN     Pastel Nata            207.0   
2           NaN   Lisboa             306.0             NaN            245.0   
3           NaN    Porto             121.0             NaN            186.0   
4           NaN   Lisboa             357.0  Pastel de Nata            186.0   
5           NaN   Lisboa             202.0  Pastel de nata            218.0   

    salt_ratio  sugar_content  vanilla_extract quality_class  
id                                                            
1        42.74           22.8              5.7            KO  
2        41.73           11.6              4.0            KO  
3        75.10           20.3              7.5            OK  
4        46.41           73.3              4.2            OK  
5        56.52           80.1              6.0            KO

In [27]:
learn_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 5200 entries, 1 to 5200
Data columns (total 17 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   ambient_humidity   5182 non-null   float64
 1   baking_duration    5199 non-null   float64
 2   cooling_period     5199 non-null   float64
 3   cream_fat_content  5176 non-null   float64
 4   egg_temperature    5176 non-null   float64
 5   egg_yolk_count     5176 non-null   float64
 6   final_temperature  5175 non-null   float64
 7   lemon_zest_ph      5174 non-null   float64
 8   notes_baker        0 non-null      float64
 9   origin             5039 non-null   object 
 10  oven_temperature   5179 non-null   float64
 11  pastry_type        3411 non-null   object 
 12  preheating_time    5181 non-null   float64
 13  salt_ratio         5187 non-null   float64
 14  sugar_content      5178 non-null   float64
 15  vanilla_extract    5182 non-null   float64
 16  quality_class      5199 non-n

In [28]:
learn_data.columns

Index(['ambient_humidity', 'baking_duration', 'cooling_period',
       'cream_fat_content', 'egg_temperature', 'egg_yolk_count',
       'final_temperature', 'lemon_zest_ph', 'notes_baker', 'origin',
       'oven_temperature', 'pastry_type', 'preheating_time', 'salt_ratio',
       'sugar_content', 'vanilla_extract', 'quality_class'],
      dtype='object')

In [29]:
learn_data.describe()

ambient_humidity  baking_duration  cooling_period  cream_fat_content  \
count       5182.000000      5199.000000     5199.000000         5176.00000   
mean          59.980316        26.979804       31.817850           98.41306   
std           11.616151         8.320079       14.427143            1.99735   
min           40.000000        14.000000        0.000000           91.90000   
25%           50.000000        21.000000       25.000000           97.10000   
50%           60.000000        25.000000       31.000000           98.40000   
75%           70.000000        30.000000       39.000000           99.80000   
max           80.000000        89.000000      105.000000          107.30000   

       egg_temperature  egg_yolk_count  final_temperature  lemon_zest_ph  \
count      5176.000000     5176.000000        5175.000000    5174.000000   
mean         70.812403       10.498068         223.577198       3.218902   
std          35.439831        1.257709          85.258779       0.165909   
min          12.000000        8.000000          31.000000       2.700000   
25%          44.000000       10.000000         164.000000       3.100000   
50%          67.000000       10.000000         226.000000       3.200000   
75%          92.000000       11.000000         284.000000       3.300000   
max         575.000000       15.000000         722.000000       4.000000   

       notes_baker  oven_temperature  preheating_time   salt_ratio  \
count          0.0       5179.000000      5181.000000  5187.000000   
mean           NaN        223.541031       216.413241    55.882034   
std            NaN         84.482034        39.533116    34.867144   
min            NaN         58.000000       114.000000    11.550000   
25%            NaN        166.000000       192.000000    37.910000   
50%            NaN        227.000000       209.000000    47.170000   
75%            NaN        283.000000       231.000000    64.610000   
max            NaN        713.000000       499.000000   640.320000   

       sugar_content  vanilla_extract  
count    5178.000000      5182.000000  
mean       54.003418         5.307159  
std        47.213038         1.483175  
min         6.000000         2.200000  
25%        18.000000         4.300000  
50%        30.000000         5.100000  
75%        80.675000         6.000000  
max       656.500000        20.800000

In [37]:
learn_data.shape

(5200, 17)

<h4>Percentage of Missing Values per column:</h4>

In [47]:
learn_data.isnull().sum()/5200*100

ambient_humidity       0.346154
baking_duration        0.019231
cooling_period         0.019231
cream_fat_content      0.461538
egg_temperature        0.461538
egg_yolk_count         0.461538
final_temperature      0.480769
lemon_zest_ph          0.500000
notes_baker          100.000000
origin                 3.096154
oven_temperature       0.403846
pastry_type           34.403846
preheating_time        0.365385
salt_ratio             0.250000
sugar_content          0.423077
vanilla_extract        0.346154
quality_class          0.019231
dtype: float64

<div style="border: 3px solid #0EE071; padding: 10px; border-radius: 6px;">
<h3>Data Structure Overview</h3>

<ul>
<li>The dataset contains <b>17 columns</b> and <b>5200 rows</b>.</li>
<li>Each row represents a <b>pastery that was baked</b>.</li>
</ul>
<br>
<h4>By colum:</h4>
<ul>
    <li><code>ambient_humidity</code>: Numerical - 0.34% missing values</li>
    <li><code>baking_duration</code>: Numerical - 0.02% missing values</li>
    <li><code>cooling_period</code>: Numerical - 0.2% missing values</li>
    <li><code>cream_fat_content</code>: Numerical - 0.46% missing values</li>
    <li><code>egg_temperature</code>: Numerical - 0.46% missing values</li>
    <li><code>egg_yolk_count</code>: Numerical - 0.46% missing values</li>
    <li><code>final_temperature</code>: Numerical - 0.48% missing values</li>
    <li><code>lemon_zest_ph</code>: Numerical - 0.5% missing values</li>
    <li><code>notes_baker</code>: <b>All Values Missings</b> - 100% missing values</li>
    <li><code>origin</code>: Object - 3.1% missing values</li>
    <li><code>oven_temperature</code>: Numerical - 0.40% missing values</li>
    <li><code>pastry_type</code>: Object - 34.40% missing values</li>
    <li><code>preheating_time</code>: Numerical - 0.36% missing values</li>
    <li><code>salt_ratio</code>: Numerical - 0.25% missing values</li>
    <li><code>sugar_content</code>: Numerical - 0.42% missing values</li>
    <li><code>vanilla_extract</code>: Numerical - 0.34% missing values</li>
    <li><code>quality_class</code>: Categorical - 0.02% missing values</li>
</ul>
</div>
<br>